lib/phlex/bucket.rb

```diff
@@ -5,38 +5,49 @@ def self.extended(mod)
 		warn "🚨 [WARNING] Phlex::Bucket is experimental and may be removed from future versions of Phlex."
 
 		# Eager load all constants in the module for apps that use Zeitwerk.
-		mod.constants.each { |c| mod.const_get(c) }
+		mod.constants.each { |c| mod.const_added(c) if mod.autoload?(c) }
 	end
 
 	def const_added(name)
-		# This can sometime be triggered by an autoload, which means it gets
-		# triggered a second time when we call `const_get` below and Ruby loads it.
-		return super if Fiber[:__phlex_adding_bucket_const__]
-
-		begin
-			Fiber[:__phlex_adding_bucket_const__] = true
-			constant = const_get(name)
-		ensure
-			Fiber[:__phlex_adding_bucket_const__] = false
-		end
-
-		if instance_methods.include?(name)
-			raise NameError, "The instance method `#{name}' is already defined on `#{inspect}`."
-		elsif methods.include?(name)
-			raise NameError, "The method `#{name}' is already defined on `#{inspect}`."
-		end
-
-		constant.include(self)
-
-		if Class === constant && constant < Phlex::SGML
+		if autoload?(name)
 			define_method(name) do |*args, **kwargs, &block|
-				render(constant.new(*args, **kwargs), &block)
+				this = method(name)
+				self.class.const_get(name) # load the constant (which should override this method)
```

Comment:
If this constant isn’t a renderable, we should probably undef this proxy method.